harness/go/harness/runner.go

```diff
@@ -247,7 +247,19 @@ func (r *Runner) CheckHistoryDefault(ctx context.Context, _ client.WorkflowRun)
 // ReplayHistories replays the given histories checking for errors.
 func (r *Runner) ReplayHistories(ctx context.Context, histories history.Histories) error {
 	// Create replayer with all the workflow funcs
-	replayer := worker.NewWorkflowReplayer()
+	var replayer worker.WorkflowReplayer
+	dataConverter := r.Feature.ClientOptions.DataConverter
+	if dataConverter != nil {
+		var err error
+		replayer, err = worker.NewWorkflowReplayerWithOptions(
+			worker.WorkflowReplayerOptions{DataConverter: dataConverter},
+		)
+		if err != nil {
+			return err
+		}
+	} else {
+		replayer = worker.NewWorkflowReplayer()
+	}
```

Comment:
```suggestion
	replayer, err = worker.NewWorkflowReplayerWithOptions(
		worker.WorkflowReplayerOptions{DataConverter: r.Feature.ClientOptions.DataConverter},
	)
	if err != nil {
		return err
	}
```

Nil option means default

Reply:
Fixed